api/integrations/slack/zulip_slack.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+#
+# slacker is a dependency for this script.
+#
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import sys
+import string
+import random
+from six.moves import range
+from typing import List, Dict
+
+import zulip
+from slacker import Slacker, Response
+
+import zulip_slack_config as config
+
+
+client = zulip.Client(email=config.ZULIP_USER, api_key=config.ZULIP_API_KEY, site=config.ZULIP_SITE)
+
+
+class FromSlackImporter(object):
+    def __init__(self, slack_token, get_archived_channels=True):
+        # type: (str, bool) -> None
+        self.slack = Slacker(slack_token)
+        self.get_archived_channels = get_archived_channels
+
+    def get_slack_users_email(self):
+        # type: () -> Dict[str, Dict[str, str]]
+
+        r = self.slack.users.list()
+        self._check_if_response_is_successful(r)
+        results_dict = {}
+        for user in r.body['members']:
+            if user['profile'].get('email') and user.get('deleted') is False:
+                results_dict[user['id']] = {'email': user['profile']['email'], 'name': user['profile']['real_name']}
+        return results_dict
+
+    def get_slack_public_channels_names(self):
+        # type: () -> List[Dict[str, str]]
+
+        r = self.slack.channels.list()
+        self._check_if_response_is_successful(r)
+        return [{'name': channel['name'], 'members': channel['members']} for channel in r.body['channels']]
+
+    def get_slack_private_channels_names(self):
+        # type: () -> List[str]
+
+        r = self.slack.groups.list()
+        self._check_if_response_is_successful(r)
+        return [
+            channel['name'] for channel in r.body['groups']
+            if not channel['is_archived'] or self.get_archived_channels
+        ]
+
+    def _check_if_response_is_successful(self, response):
+        # type: (Response) -> None
+        print(response)
+        if not response.successful:
+            print(response.error)
+            sys.exit(1)
+
+def _generate_random_password(size=10):
+    # type: (int) -> str
+    return ''.join(random.choice(string.ascii_letters + string.digits) for _ in range(size))
+
+def get_and_add_users(slack_importer):
+    # type: (Slacker) -> Dict[str, Dict[str, str]]
+    users = slack_importer.get_slack_users_email()
+    added_users = {}
+    print('######### IMPORTING USERS STARTED #########\n')
+    for user_id, user in users.items():
+        r = client.create_user({
+            'email': user['email'],
+            'password': _generate_random_password(),
```

Comment:
Is the password argument required?  Might be simpler to create them with no user.